### PR TITLE
Add `Blind_call` call type

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR adds the `Blind_call` call type. To compare the call semantics of this with the others:
* `Call` repesents a plain call, conveying permission to the child (and potentially it's children, via delegate call) to use the `token_id` derived from the caller's account ID
* `Delegate_call` represents a call as if this node was not in the stack, passing permission to use the `token_id` given to this node down to its child
* `Blind_call` is a call that conveys no permission to its children to use its `token_id`
  - the current proposed semantics are that this also resets the stack of callers visible to the zkApp. Do we also need a `No_token_call` that also doesn't pass `token_id`, but *doesn't* reset the stack?

This builds upon #12450.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by psteckler** [comment]
Can the above description be added to where the data type is defined in the code, for easy future reference?

**Review by psteckler — CHANGES_REQUESTED** [review]
Please add the description of the intended semantics where the call type is defined.

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by psteckler — APPROVED** [review]
one nit, otherwise good

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/mina_base/account_update.ml [hunk 1]
```diff
@@ -352,39 +374,76 @@ module Call_type = struct
...
+              let open Checked in
+              let%bind () = typ.check x in
+              let sum =
+                Field.Var.(add (is_delegate_call :> t) (is_blind_call :> t))
+              in
+              (* Assert boolean; we should really have a helper for this
+                 somewhere.
+              *)
+              let%bind sum_squared = Field.Checked.mul sum sum in
+              Field.Checked.Assert.equal sum sum_squared )
```
Comment (mitschabaude) [inline src/lib/mina_base/account_update.ml:408]: I guess this `check` runs in addition to the check that the two fields are booleans individually?

Comment (mitschabaude) [reply src/lib/mina_base/account_update.ml:408]: ah I guess that's the `typ.check x` above

Comment (mitschabaude) [reply src/lib/mina_base/account_update.ml:408]: 👍🏻

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/mina_ledger/ledger.ml [hunk 2]
```diff
@@ -539,10 +539,10 @@ let%test_unit "tokens test" =
     let token_minting =
       mk_forest
         [ mk_node
-            (mk_account_update_body Signature Call token_owner Token_id.default
+            (mk_account_update_body Signature No token_owner Token_id.default
                (-account_creation_fee) )
             [ mk_node
-                (mk_account_update_body None_given Call token_account1
+                (mk_account_update_body None_given No token_account1
```
Comment (deepthiskumar) [inline src/lib/mina_ledger/ledger.ml:545]: This should be `Parents_own_token`, right? `custom_token_id` is owned by `token_owner`

File: src/lib/mina_ledger/ledger.ml [hunk 3]
```diff
@@ -551,30 +551,30 @@ let%test_unit "tokens test" =
     let token_transfers =
       mk_forest
         [ mk_node
-            (mk_account_update_body Signature Call token_owner Token_id.default
+            (mk_account_update_body Signature No token_owner Token_id.default
                (-account_creation_fee) )
             [ mk_node
-                (mk_account_update_body Signature Call token_account1
+                (mk_account_update_body Signature No token_account1
```
Comment (deepthiskumar) [inline src/lib/mina_ledger/ledger.ml:557]: Here as well and below where ever `custom_token_id` is used

File: src/app/test_executive/zkapps.ml [hunk 4]
```diff
@@ -421,11 +421,11 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
         let with_dummy_signatures =
           mk_forest
             [ mk_node
-                (mk_account_update_body Signature Call token_owner
+                (mk_account_update_body Signature No token_owner
                    Token_id.default
                    (-account_creation_fee_int) )
                 [ mk_node
-                    (mk_account_update_body Signature Call token_accounts.(0)
+                    (mk_account_update_body Signature No token_accounts.(0)
```
Comment (deepthiskumar) [inline src/app/test_executive/zkapps.ml:428]: This should be `Parents_own_token` as well

File: src/app/test_executive/zkapps.ml [hunk 5]
```diff
@@ -440,15 +440,15 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
         let with_dummy_signatures =
           mk_forest
             [ mk_node
-                (mk_account_update_body Signature Call token_owner
+                (mk_account_update_body Signature No token_owner
                    Token_id.default
                    (-2 * account_creation_fee_int) )
                 [ mk_node
-                    (mk_account_update_body Signature Call token_owner
+                    (mk_account_update_body Signature No token_owner
```
Comment (deepthiskumar) [inline src/app/test_executive/zkapps.ml:447]: `Parents_own_token`

File: src/app/test_executive/zkapps.ml [hunk 6]
```diff
@@ -440,15 +440,15 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
...
                        custom_token_id 0 )
                     [ mk_node
-                        (mk_account_update_body Signature Call
-                           token_accounts.(2) custom_token_id2 500 )
+                        (mk_account_update_body Signature No token_accounts.(2)
```
Comment (deepthiskumar) [inline src/app/test_executive/zkapps.ml:450]: `Parents_own_token`

File: src/app/test_executive/zkapps.ml [hunk 7]
```diff
@@ -464,23 +464,23 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
         let with_dummy_signatures =
           mk_forest
             [ mk_node
-                (mk_account_update_body Signature Call token_owner
+                (mk_account_update_body Signature No token_owner
                    Token_id.default
                    (-account_creation_fee_int) )
                 [ mk_node
-                    (mk_account_update_body Signature Call token_accounts.(0)
+                    (mk_account_update_body Signature No token_accounts.(0)
```
Comment (deepthiskumar) [inline src/app/test_executive/zkapps.ml:471]: `Parents_own_token` everywhere in this txn where `custom_token_id` is used

File: src/app/test_executive/zkapps.ml [hunk 8]
```diff
@@ -496,27 +496,27 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
         let with_dummy_signatures =
           mk_forest
             [ mk_node
-                (mk_account_update_body Signature Call token_owner
+                (mk_account_update_body Signature No token_owner
                    Token_id.default
                    (-account_creation_fee_int) )
                 [ mk_node
-                    (mk_account_update_body Signature Call token_accounts.(1)
+                    (mk_account_update_body Signature No token_accounts.(1)
```
Comment (deepthiskumar) [inline src/app/test_executive/zkapps.ml:503]: `Parents_own_token` everywhere in this txn where `custom_token_id` or `custom_token_id2` is used

**Review by joaosreis — APPROVED** [review]

**Pull request merged**